### PR TITLE
Add observation table joins

### PR DESCRIPTION
**Pull request by ssoper-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: src/test/java/gov/usgs/wma/waterdata/LoadTimeSeriesTest.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
+
+import javax.naming.spi.DirStateFactory;
```
Comment (ssoper-usgs) [inline src/test/java/gov/usgs/wma/waterdata/LoadTimeSeriesTest.java]: whoops, will delete this.

File: src/test/java/gov/usgs/wma/waterdata/LoadTimeSeriesTest.java [hunk 2]
```diff
@@ -35,35 +36,74 @@ public void setupLoadTimeSeries() {
...
+		}, "should have thrown an exception but did not");
 	}
 
+	// tests for successes
 	@Test
 	public void testFoundGeneric() {
 		genericTimeSeriesList.add(genericTimeSeries1);
 		genericTimeSeriesList.add(genericTimeSeries2);
-		// happy path - 2 time series returned
+		// 2 time series returned
```
Comment (dsteinich) [inline src/test/java/gov/usgs/wma/waterdata/LoadTimeSeriesTest.java]: Technically, aren't these time steps within a single time series. It is a nit, but we need to be as clear as possible so the next person doesn't get confused.

File: src/main/java/gov/usgs/wma/waterdata/LoadTimeSeries.java [hunk 3]
```diff
@@ -15,6 +15,9 @@
 
 	public static final String STATUS_SUCCESS = "success";
 	public static final String STATUS_FAIL = "fail";
+	public static final String FAIL_MESSAGE_NO_RECORDS = "No records found";
```
Comment (dsteinich) [inline src/main/java/gov/usgs/wma/waterdata/LoadTimeSeries.java]: Bonus points, have the message here match the logged message. String.format() could be used to get the dynamic values into the message. https://www.baeldung.com/string/format

File: src/main/resources/sql/insertTimeSeries.sql [hunk 4]
```diff
@@ -24,14 +24,14 @@ insert into
 values (
 	?
 	,?
-	,NULL /*(select monitoring_location_id from monitoring_location where monitoring_location_identifier = ?)*/
+	,(select monitoring_location_id from monitoring_location where monitoring_location_identifier = ?)
 	,?
 	,?
-	,NULL
-	,NULL
+	,(select parm_nm from parm where parm_cd = ?)
```
Comment (dsteinich) [inline src/main/resources/sql/insertTimeSeries.sql:30]: Not sure these are correct, the references should be a URL? (both parm & stat).

See example in https://internal.cida.usgs.gov/jira/browse/IOW-135

We may also need to do this insert differently if performance becomes as issue. Selecting twice from the same table is never as performant as doing it once. (Or in the case of the current query twice per inserted row.)

Comment (ssoper-usgs) [reply src/main/resources/sql/insertTimeSeries.sql:30]: I'm not sure about a url - there do not appear to be urls in any column in either the stat or parm tables... should we be creating these?

Comment (ssoper-usgs) [reply src/main/resources/sql/insertTimeSeries.sql:30]: Ah, I think I see what you mean.  We can concat the url "https://waterdata.usgs.gov/nwisweb/rdf?parmCd=" with the parm code for observed property reference, and we can concat "https://waterdata.usgs.gov/nwisweb/rdf?statCd=" with the stat code for statistic reference.

**Pull request merged**